### PR TITLE
add REUSE.software license information

### PR DESCRIPTION
**Pull request by fneum**
Closes #65.

Added [FSFE REUSE](https://reuse.software) compliant license information. Documentation now licensed under CC-BY-4.0

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -184,10 +193,15 @@ The included ``.nc`` files are PyPSA network files which can be imported with Py
...
 
-The code in PyPSA-Eur is released as free software under the `GPLv3
-<http://www.gnu.org/licenses/gpl-3.0.en.html>`_, see
-`LICENSE <https://github.com/PyPSA/pypsa-eur/blob/master/LICENSE.txt>`_.
-However, different licenses and terms of use apply to the various input data, which are summarised below.
+PyPSA-Eur work is released under multiple licenses:
+
+* All original source code is licensed as free software under `GPL-3.0-or-later <LICENSES/GPL-3.0-or-later.txt>`_.
+* The documentation is licensed under `CC-BY-4.0 <LICENSES/CC-BY-4.0.txt>`_.
+* Configuration and data files are mostly licensed under `CC0-1.0 <LICENSES/CC0-1.0.txt>`_.
```
Comment (euronion) [inline doc/index.rst:200]: According to the [dep5 file](https://github.com/PyPSA/pypsa-eur/blob/license/.reuse/dep5) all files in `data` are licenses as CC-BY-4.0.
How about separating it?

```suggestion
* Configuration files are mostly licensed under `CC0-1.0 <LICENSES/CC0-1.0.txt>`_.
* Data files are licensed under `CC-BY-4.0 <LICENSES/CC-BY-4.0.txt>`_.
```

File: doc/index.rst [hunk 2]
```diff
@@ -184,10 +193,15 @@ The included ``.nc`` files are PyPSA network files which can be imported with Py
...
+
+See the individual files for license details.
```
Comment (euronion) [inline doc/index.rst:202]:
```suggestion
See the individual files and the `dep5 <.reuse/dep5>`_ file for license details.
```

**Comment by euronion** [comment]
quick x-check for completness:

```
pypsa-eur>reuse lint
# SUMMARY

* Bad licenses:
* Deprecated licenses:
* Licenses without file extension:
* Missing licenses:
* Unused licenses:
* Used licenses: CC-BY-4.0, CC0-1.0, GPL-3.0-or-later
* Read errors: 0
* Files with copyright information: 152 / 152
* Files with license information: 152 / 152

Congratulations! Your project is compliant with version 3.0 of the REUSE Specification :-)
```

**Comment by fneum** [comment]
CI failure on Windows only due to separate issue. Merge anyway with "[skip travis]". Opened dedicated issue #161.

```
Done. Your build exited with 1.

Traceback (most recent call last):

  File "C:\miniconda\envs\pypsa-eur\lib\site-packages\snakemake\__init__.py", line 699, in snakemake

    keepincomplete=keep_incomplete,

  File "C:\miniconda\envs\pypsa-eur\lib\site-packages\snakemake\workflow.py", line 632, in execute

    or delete_temp_output,

  File "C:\miniconda\envs\pypsa-eur\lib\site-packages\snakemake\persistence.py", line 72, in __init__

    os.makedirs(d, exist_ok=True)

  File "C:\miniconda\envs\pypsa-eur\lib\os.py", line 223, in makedirs

    mkdir(name, mode)

NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\travis\\build\\PyPSA\\pypsa-eur\\.snakemake\\aux'
```

File: doc/release_notes.rst [hunk 3]
```diff
@@ -11,6 +16,8 @@ PyPSA-Eur 0.2.0 (TBD)
 
 * ``snakemake`` rules for retrieving cutouts and the natura raster can now be disabled independently from their respective rules to build them; via ``config.*yaml`` (`#136 <https://github.com/PyPSA/pypsa-eur/pull/136>`_).
 
+* Added `FSFE REUSE <https://reuse.software>`_ compliant license information. Documentation now licensed under CC-BY-4.0 (`#159 <https://github.com/PyPSA/pypsa-eur/pull/159>`_).
```
Comment (fneum) [inline doc/release_notes.rst:19]:
```suggestion
* Added `FSFE REUSE <https://reuse.software>`_ compliant license information. Documentation now licensed under CC-BY-4.0 (`#160 <https://github.com/PyPSA/pypsa-eur/pull/160>`_).
```

**Pull request merged**